graphql_compiler/macros/macro_edge/helpers.py

```diff
@@ -121,3 +121,48 @@ def remove_directives_from_ast(ast, directive_names_to_omit):
     new_ast.selection_set = SelectionSet(new_selections)
     new_ast.directives = directives_to_keep
     return new_ast
+
+
+def omit_ast_from_ast_selections(ast, ast_to_omit):
+    """Return an equivalent AST to the input, but with the specified AST omitted if it appears.
```

Comment:
maybe clarify that you look for literally the same AST object

Reply:
The `ast_to_omit` argument definition already says that, would you prefer I include it elsewhere?
```
ast_to_omit: GraphQL library AST object, the *exact same* object that should be omitted.
             This function uses reference equality, since deep equality can get expensive.
```